test/extensions/http/cache/file_system_http_cache/cache_file_fixed_block_test.cc

```diff
@@ -0,0 +1,96 @@
+#include "source/extensions/http/cache/file_system_http_cache/cache_file_fixed_block.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+namespace FileSystemHttpCache {
+
+class CacheFileFixedBlockTest : public ::testing::Test {
+public:
+  // Wrappers for private test-only functions.
+  void setFileId(CacheFileFixedBlock& block, uint32_t id) { block.setFileId(id); }
+  void setCacheVersionId(CacheFileFixedBlock& block, uint32_t id) { block.setCacheVersionId(id); }
+};
+
+namespace {
+
+TEST_F(CacheFileFixedBlockTest, InitializesToValid) {
+  CacheFileFixedBlock default_block;
+  EXPECT_TRUE(default_block.isValid());
+}
+
+TEST_F(CacheFileFixedBlockTest, GettersRecoverValuesThatWereSet) {
+  CacheFileFixedBlock block;
+  setFileId(block, 98765);
+  setCacheVersionId(block, 56789);
+  block.setBodySize(999999);
+  block.setHeadersSize(1234);
+  block.setTrailersSize(4321);
```

Comment:
I guess this makes sense to test this way as your getters do real work.

My suggestion would be to instead do the work on read/write from files and make the getter trivial (expressed in impl).

But in either case I think you could probably make a second CacheFileFixedBlock and populate it by reading from it 'block' and writing to the new one, and *then* checking the getters.

Reply:
Doing the work on read/write would mean allocating a new buffer into which to write the byte-reordered values. The whole point of this object is to be that serialized buffer - creating a separate buffer for serialize/deserialize wouldn't really make sense.

There is a test case that validates values (via offsets) after a serialize/deserialize, but having that test and not this test would not reveal a misbehavior in which, say, 'set' flips the byte order, one of serialize or deserialize flips the byte order, and 'get' does not. Which is the sort of misbehavior you might get if someone decided doing the work on read/write was better and then made that change poorly!

Reply:
understood -- I think it's better to answer the question fo why you need to be hacking network byte-order in your getter as opposed to in your serialzier/deserializer.

Let's resolve that first.

Reply:
Because changing the byte order in the serialize/deserialize operation would mean you need *another* buffer that you're serializing or deserializing into. The whole point of this class is to represent a serialized buffer.
(For 'read' it's not important because it's already reading from an existing buffer, but for 'write' you'd need the caller to provide a memcpy-recipient pointer pointing at a sufficiently sized block of data, which doesn't play well with Envoy::Buffer or safety, OR provide an Envoy::Buffer& which doesn't play well with being able to do anything else.)

Reply:
This makes sense, but comments in the code would help. I'd like another opinion about it, as I think it's a bit unusual to do it this way. I'll pull someone in once there are comments explaining this decision in the code.

Reply:
Hideousness now reduced with helper functions.